MariaDB/server
# MDEV-33492: mysql_install_db fails when baseurl is set

- [x] *The Jira issue number for this PR is: MDEV-33492*


## Description

auth_map.so isn't guaranteed to be available. Fedora packages it separately.

The --base-dir path of mysql_install_db.sh seems to contain historicial heuristics that have been replaced on other branches of the script.

We attempt to do the same here placing the basedir original paths so that all components are absolute.

## Release Notes
TODO: What should the release notes say about this change?
Include any changed system variables, status variables or behaviour. Optionally list any https://mariadb.com/kb/ pages that need changing.

## How can this PR be tested?

```
$ cmake --install . --prefix /tmp/m10.11
$ rm -rf /tmp/dd-10.11/
$ mkdir /tmp/dd-10.11
$ /tmp/m10.11/scripts/mariadb-install-db --no-defaults --basedir=/tmp/m10.11 --datadir=/tmp/dd-10.11 --verbose
Installing MariaDB/MySQL system tables in '/tmp/dd-10.11' ...
2024-02-21 17:16:54 0 [Note] Starting MariaDB 10.11.8-MariaDB source revision 32f535d45c7cf1ef5e445e7814dbedc67b23b7b7 as process 164589
2024-02-21 17:16:54 0 [Note] InnoDB: The first data file './ibdata1' did not exist. A new tablespace will be created!
2024-02-21 17:16:54 0 [Note] InnoDB: Compressed tables use zlib 1.2.13
2024-02-21 17:16:54 0 [Note] InnoDB: Number of transaction pools: 1
2024-02-21 17:16:54 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
2024-02-21 17:16:54 0 [Note] InnoDB: Using liburing
2024-02-21 17:16:54 0 [Note] InnoDB: Initializing buffer pool, total size = 128.000MiB, chunk size = 2.000MiB
2024-02-21 17:16:54 0 [Note] InnoDB: Initialized memory pressure event listener
2024-02-21 17:16:54 0 [Note] InnoDB: Completed initialization of buffer pool
2024-02-21 17:16:54 0 [Note] InnoDB: Setting file './ibdata1' size to 12.000MiB. Physically writing the file full; Please wait ...
2024-02-21 17:16:54 0 [Note] InnoDB: File './ibdata1' size is now 12.000MiB.
2024-02-21 17:16:54 0 [Note] InnoDB: Buffered log writes (block size=512 bytes)
2024-02-21 17:16:55 0 [Note] InnoDB: Doublewrite buffer not found: creating new
2024-02-21 17:16:55 0 [Note] InnoDB: 128 rollback segments are active.
2024-02-21 17:16:55 0 [Note] InnoDB: Setting file './ibtmp1' size to 12.000MiB. Physically writing the file full; Please wait ...
2024-02-21 17:16:55 0 [Note] InnoDB: File './ibtmp1' size is now 12.000MiB.
2024-02-21 17:16:55 0 [Note] InnoDB: log sequence number 0; transaction id 3
OK

To start mariadbd at boot time you have to copy
support-files/mariadb.service to the right place for your system


Two all-privilege accounts were created.
One is root@localhost, it has no password, but you need to
be system 'root' user to connect. Use, for example, sudo mysql
The second is dan@localhost, it has no password either, but
you need to be the system 'dan' user to connect.
After connecting you can set the password, if you would need to be
able to connect as any of these users with a password and without sudo

See the MariaDB Knowledgebase at https://mariadb.com/kb

You can start the MariaDB daemon with:
cd '/tmp/m10.11' ; /tmp/m10.11/bin/mariadbd-safe --datadir='/tmp/dd-10.11'

You can test the MariaDB daemon with mysql-test-run.pl
cd '/tmp/m10.11/mysql-test' ; perl mariadb-test-run.pl

Please report any problems at https://mariadb.org/jira

The latest information about MariaDB is available at https://mariadb.org/.

Consider joining MariaDB's strong and vibrant community:
https://mariadb.org/get-involved/

```

TODO: modify the automated test suite to verify that the PR causes MariaDB to behave as intended.
Consult the documentation on ["Writing good test cases"](https://mariadb.org/get-involved/getting-started-for-developers/writing-good-test-cases-mariadb-server).


If the changes are not amenable to automated testing, please explain why not and carefully describe how to test manually.


## Basing the PR against the correct MariaDB version


## PR quality check
